core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -71,6 +72,16 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val actorSyste
   // Executors we have requested the cluster manager to kill that have not died yet
   private val executorsPendingToRemove = new HashSet[String]
 
+  /**
+   * Send new credentials to executors. This is the method that is called when the scheduled
+   * login completes, so the new credentials can be sent to the executors.
+   * @param credentials
+   */
+  def sendNewCredentialsToExecutors(credentials: SerializableBuffer): Unit = {
+    // We don't care about the reply, so going to deadLetters is fine.
```

Comment:
BTW, on top of that, the executor should not service any tasks before it downloads the credentials, so it either should request them before registering, or some other form of blocking should be added.